.github/git-release-tool/git-release-tool.sh

```diff
@@ -0,0 +1,347 @@
+
+# Helper functions
+## Logging functions
+log_notice() {
+  echo "[Notice] "$1
+}
+
+log_error() {
+  if [[ "$LOG_VERBOSITY_NUMBER" -ge 0 ]]; then
+    echo "[Error] "$1 >/dev/stderr
+  fi
+}
+
+log_info() {
+  if [[ "$LOG_VERBOSITY_NUMBER" -ge 1 ]]; then
+    echo "[Info] "$1
+  fi
+}
+
+log_debug() {
+  if [[ "$LOG_VERBOSITY_NUMBER" -ge 2 ]]; then
+    echo "[Debug] "$1
+  fi
+}
+
+## Usage info
+usage() {
+  echo
+  log_notice "$0 - script to cut releases for TIP OpenWIFI CloudSDK 2.x repos"
+  log_notice
+  log_notice "This script requires configuration file 'repositories.yaml' near the script and list of environment variables to work"
+  log_notice
+  log_notice "repositories.yaml file format:"
+  echo "deploy_repo_url: git@github.com:Telecominfraproject/wlan-cloud-ucentral-deploy.git # modify if repo name changes"
+  echo "repositories:"
+  echo "  - name: owgw-ui # should be the same as in image repository in helm values (i.e. tip-tip-wlan-cloud-ucentral.jfrog.io/owgw-ui)"
+  echo "    url: git@github.com:Telecominfraproject/wlan-cloud-owprov-ui.git # it's up to you to use SSH or HTTPS format and setup credentials for push/pull"
+  echo "    docker_compose_name: OWPROVUI # name of environment variable in docker-compose .env file containing image tag for the service"
+  log_notice
+  log_notice "List of required environment variables:"
+  log_notice "- RELEASE_VERSION - release version that should be applied to repositories. Should comply release nameing policy (valid example - 'v2.0.0')"
+  log_notice "- TAG_TYPE - type of tag that should be created for release (supported values - RC / FINAL)"
+  log_notice "- GIT_PUSH_CONFIRMED - confirmation that any changes should be pushed to git (dry-run if unset, set to 'true' to enable)"
+  log_notice
+  log_notice "You may increase log verbosity by setting environment variable LOG_VERBOSITY to required level (ERROR/INFO/DEBUG)"
+#
+}
+
+## Setting functions
+set_log_verbosity_number() {
+  # Log verbosity levels:
+  # 0 - ERROR
+  # 1 - INFO
+  # 2 - DEBUG
+  case $LOG_VERBOSITY in
+    ERROR )
+      LOG_VERBOSITY_NUMBER=0
+    ;;
+    INFO )
+      LOG_VERBOSITY_NUMBER=1
+    ;;
+    DEBUG )
+      LOG_VERBOSITY_NUMBER=2
+    ;;
+    * )
+      log_notice "Setting LOG_VERBOSITY to INFO by default"
+      LOG_VERBOSITY_NUMBER=1
+    ;;
+  esac
+}
+
+## Git manipulation functions
+modify_deploy_repo_values() {
+  NEW_RELEASE_TAG=$1
+  REPOSITORIES_AMOUNT=$(cat ../repositories.yaml | yq ".repositories[].name" -r | wc -l)
+  for REPO_INDEX in $(seq 0 $(expr $REPOSITORIES_AMOUNT - 1)); do
+    REPO_URL=$(cat ../repositories.yaml | yq ".repositories[$REPO_INDEX].url" -r)
+    REPO_URL_WITHOUT_SUFFIX=${REPO_URL%.git}
+    REPO_DOCKER_COMPOSE_NAME=$(cat ../repositories.yaml | yq ".repositories[$REPO_INDEX].docker_compose_name" -r)
+    SERVICE_TAG="${REPO_TAGS_ARRAY[$REPO_INDEX]}"
+    sed "s/$REPO_DOCKER_COMPOSE_NAME=.*/$REPO_DOCKER_COMPOSE_NAME=$SERVICE_TAG/" -i docker-compose/.env
```

Comment:
there are two more files which have to be adapted, `.env.letsencrypt` and `.env.selfsigned`.

Reply:
Sure, thanks for noticing!